OpenSim/Sandbox/MatlabScripts/muscleCurves.m

```diff
@@ -0,0 +1,264 @@
+end
+
+muscle = struct();
+% Cycle through each available coordinate and save its range values.
+% These will get used later to run calculate muscle force on each
+% coordinate value.
+for u = 1 : length(muscCoord)
+    % Get a reference to the coordinate
+    aCoord = model.getCoordinateSet.get(muscCoord(u));
+    % Create an arrary of radian value's for the range
+    coordRange = (aCoord.getRangeMin:0.01:aCoord.getRangeMax)';
+    % add the coordinate's and their range values to the structure
+    eval(['muscle.coordinates.' char(model.getCoordinateSet.get(muscCoord(u))) ' = [coordRange];' ])
+
+end
+muscle.name = muscleName;
+end
+
+
+function [fl_active,fl_passive] = getForceLength(model, s, muscle)
+
+import org.opensim.modeling.*      % Import OpenSim Libraries
+
+coordNames = fieldnames(muscle.coordinates);
+% get the number of coordinates for the muscle
+nCoords = length( coordNames );
+
+% Get the muscle that is needed
+force = model.getMuscles().get(muscle.name);
+% Get the muscleType of that force
+muscleType = char(force.getConcreteClassName);
+% Get a reference to the concrete muscle class in the model
+eval(['myMuscle =' muscleType '.safeDownCast(force);'])
+% Display the muscle name
+%display(char(myMuscle))
+
+% matrix for storing the total complete fl curve
+flMatrix = zeros(2,3);
+
+for k = 1 : nCoords
+
+   % Get the name of the coordinate
+   aCoord = model.getCoordinateSet.get( char(coordNames(k)) );
+   updCoord = model.updCoordinateSet.get( char(coordNames(k)) );
+   coordRange = muscle.coordinates.(coordNames{k});
+   storageData = zeros( length(coordRange), 5 );
+
+
+       % Loop through each coordinate value and get the fibre
+       % length and force of the muscle.
+       for j = 1 : length( coordRange )
+
+            % Get a current coordinate value
+            coordValue = coordRange(j);
+            updCoord.setValue(s, coordValue);
+            updCoord.setSpeedValue(s, 0 );
+
+            % Set the activation and fiber length
+            myMuscle.setActivation( s, 1 )
+            myMuscle.setDefaultFiberLength( 0.01 )
+            myMuscle.setFiberLength( s, myMuscle.getOptimalFiberLength )
+            % Equilibrate the forces from the activation
+            model.equilibrateMuscles( s );
+
+            % Store all the data in the zero matrix
+            storageData(j,:) = [...
+                rad2deg(coordValue) ...                        % Coordinate Value
+                myMuscle.getFiberLength(s) ...             % Fiber length
+                myMuscle.getNormalizedFiberLength(s) ...   % Normalized Fibre Length
+                myMuscle.getActiveFiberForce(s) ...        % Active Force
+                myMuscle.getPassiveFiberForce(s) ];        % passive fibre forces
+
+                % check to see if that the fiber length has already been se
+            if isempty( find( myMuscle.getNormalizedFiberLength(s) == flMatrix(:,1), 1 ) )
+                flMatrix = [flMatrix ; storageData(j,3:5)];
```

Comment:
If only columns 3, 4, and 5 are appended to `flMatrix`, perhaps there is no need to store columns 1 (coordinate value) or 2 (fiber length).